libs/@local/harpc/tower/src/body/stream.rs

```diff
@@ -0,0 +1,227 @@
+use core::task::ready;
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use bytes::Buf;
+use futures::Stream;
+
+use super::{Body, BodyState, Frame};
+
+pin_project_lite::pin_project! {
+    #[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    pub struct StreamBody<S> {
+        #[pin]
+        stream: S,
+        is_complete: bool,
```

Comment:
I wonder if we can use `Option<S>` instead of an additional boolean

Reply:
done in [a2b9b20](https://github.com/hashintel/hash/pull/4713/commits/a2b9b2004318dbfcfc2af3ed54689d528624e4b8)